docs/spec/reactors/consensus/proposer-selection.md

```diff
@@ -2,20 +2,228 @@
+ 
+
+|Priority Round |-3 | -2 | -1 | 0  | 1   | 2   | 3 | 4 | 5 | Comment
+|---------------|-- | ---|--- |--- |---  |---  |---|---|---|--------
+| R end         |p3 |    |    |    | p1  | p2  |   |   |   |remove p2
+| 1             |   |    |    | p3 |     | p1  |   |   |   |A(i)+=VP(i)
+|               |   | p1 |    | p3 |     |     |   |   |   |A(p1)-= P
+| *new step*    |   |    | p1 |    | p3  |     |   |   |   |A(i) -= avg
+
+The procedure could continue without modifications. However, it is possible that after a sufficiently large number of modifications in validator set, the priority values would group towards maximum or minimum allowed values.
+For this reason, the selection procedure adds another step (see last row) that shifts the current priority values left or right such that the average remains 0.
+
+The modified selection algorithm is:
+
+    def singleProposalSelection (vset):
+        for each validator i in vset:
+            A(i) += VP(i)
+        prop = max(A)
+        A(prop) -= P
+    
+    def proposalSelection (vset, times):
+        for in in range(times):
+        singleProposalSelection(vset)
+
+        // normalize - shift with average priority
+        avg = sum(A(i) for i in vset)/len(vset)
+        for each validator i in vset:
+            A(i) -= avg
+
+Note that the shifting operation happens after singleProposalSelection() has run times times.
+
+#### New Validator
+When a new validator is added same problem as the one described for removal appears, the sum of priorities in the new set is not zero. This is now fixed with the shift step introduced above.
+
+One other issue that needs to be addressed is the following. A validator V with low voting power that has just been elected is moved to the end of the queue. If the validator set is large and/ or with other validators with significantly higher power, validator V will have to wait a lot of rounds to be elected. If V could remove and read itself to the validator set, it would make a significant (albeit unfair) "jump" ahead in the queue, decreasing the numbers of rounds it has to wait. 
+
+In order to prevent this, when a new validator is added, its initial priority is not set to 0 but to a lower value. This value is determined from the simulation of the above scenario: it is assumed that V had just proposed a block and was at the back of the queue when it was removed and re-added. In addition, a penalty factor is applied.
+
+If `vset = {v1,..,vn, V}` and `V` would be selected as proposer then it would have gone through the following changes:
+
+    ...
+      A(V)+=VP(V) // part of first for loop
+    ...
+    A(V)-=P  
+
+It follows that `A(V)-=sum(VP(i) for i in {v1,..,vn})`
+
+Therefore when `V` is added to the `{v1,..,vn}` set its initial priority will be:
+
+    A(V) = -sum(VP(i) for i in {v1,..,vn})
+
+With a penalty factor of 1.125, the initial priority for a newly added verifier is:
```

Comment:
Can we explain why we need the penalty factor and why `9/8` is reasonable?

Reply:
Will try my best, not sure why the 1/8th was chosen though.

Reply:
please check the latest to see if your comment is addressed